pkg/kubelet/kubelet_node_status.go

```diff
@@ -382,6 +382,8 @@ func (kl *Kubelet) syncNodeStatus() {
 		return
 	}
 	if kl.registerNode {
+		// When kubelet start first time, ensure runtime state is in real state.
+		kl.updateRuntimeUp()
```

Comment:
just to clarify, we already tell the kubelet to update its runtime up status in the kubelet run() invocation after we start the go routine for the node lease controller.  

see:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1397

it's not clear to me why this is needed.

Reply:
But we should ensure kl.updateRuntimeUp() exec **BEFORE** kl.registerWithAPIServer(). Or it will report wrong status of runtime. We can not guarantee  kl.updateRuntimeUp() exec before  kl.registerWithAPIServer() if only run kl.updateRuntimeUp() in a goroutine.

Reply:
> But we should ensure kl.updateRuntimeUp() exec **BEFORE** kl.registerWithAPIServer(). Or it will report wrong status of runtime. We can not guarantee kl.updateRuntimeUp() exec before kl.registerWithAPIServer() if only run kl.updateRuntimeUp() in a goroutine.

But we should ensure kl.updateRuntimeUp() exec BEFORE kl.registerWithAPIServer(). Or it will report wrong status of runtime. We can not guarantee kl.updateRuntimeUp() exec before kl.registerWithAPIServer() if only run kl.updateRuntimeUp() in a goroutine.

Reply:
> just to clarify, we already tell the kubelet to update its runtime up status in the kubelet run() invocation after we start the go routine for the node lease controller.
> 
> see:
> https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1397
> 
> it's not clear to me why this is needed.

@derekwaynecarr 
![image](https://user-images.githubusercontent.com/16590851/78853325-321e2f00-7a51-11ea-888a-18633fb6a622.png)
10.0.0.194 NotReady beasuse I restart kubelet. 
There is a small probability of recurrence it after 1.14.
And it must reappear before 1.14.
![image](https://user-images.githubusercontent.com/16590851/78853464-80cbc900-7a51-11ea-9955-687e2b8d560c.png)

Reply:
@mattjmcnaughton 
@derekwaynecarr